src/AggregateFunctions/AggregateFunctionOrderBy.h

```diff
@@ -0,0 +1,214 @@
+
+public:
+    AggregateFunctionOrderBy(AggregateFunctionPtr nested, const DataTypes & types, const Array & params_)
+        : IAggregateFunctionHelper<AggregateFunctionOrderBy>(types, params_)
+        , nested_func(nested), num_arguments(types.size())
+    {
+        if (params_.size() == 0)
+            throw Exception("Aggregate function " + getName() + " require at least one parameter", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        if (params_[0].getType() != Field::Types::Which::String)
+            throw Exception("First parameter for aggregate function " + getName() + " must be String", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        param_str = params_[0].get<String>();
+        types_.reserve(types.size());
+        for (size_t i = 0; i != types.size(); ++i) {
+            types_.push_back(types[i]->getPtr());
+        }
+    }
+
+    String getName() const override
+    {
+        return nested_func->getName() + "OrderBy";
+    }
+
+    DataTypePtr getReturnType() const override
+    {
+        return nested_func->getReturnType();
+    }
+
+    void create(AggregateDataPtr place) const override
+    {
+        new(place) OrderByData(param_str, types_);    
+        nested_func->create(place + sizeof(OrderByData));
+    }
+
+    void destroy(AggregateDataPtr place) const noexcept override {
+        reinterpret_cast<OrderByData *>(place)->~OrderByData();
+        nested_func->destroy(place + sizeof(OrderByData));
+    }
+
+    size_t sizeOfData() const override
+    {
+        return sizeof(OrderByData) + nested_func->sizeOfData();
+    }
+
+    size_t alignOfData() const override
+    {
+        return alignof(OrderByData);
+    }
+
+    bool hasTrivialDestructor() const override {
+        return ((std::is_trivially_destructible_v<OrderByData>) && (nested_func->hasTrivialDestructor()));
+    }
+
+    void merge(AggregateDataPtr __restrict place, ConstAggregateDataPtr rhs, Arena *) const override 
+    {
+        for (size_t i = 0; i != reinterpret_cast<const OrderByData *>(rhs)->data[0]->size(); ++i) { 
+            for (size_t j = 0; j != reinterpret_cast<const OrderByData *>(rhs)->data.size(); ++j) {
+                reinterpret_cast<OrderByData *>(place)->data[j]->insertFrom(*(reinterpret_cast<const OrderByData *>(rhs)->data[j]), i);
+            }         
+        }
+    }
+
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t>) const override 
+    {
+        size_t column_size = reinterpret_cast<const OrderByData *>(place)->data[0]->size();
+        writeBinary(column_size, buf);
+        auto size = reinterpret_cast<const OrderByData *>(place)->data.size();
+        for (size_t i = 0; i != size; ++i) {
+            for (size_t j = 0; j != column_size; ++j) {
+                auto ob_data = reinterpret_cast<const OrderByData *>(place);
+                writeBinary((*ob_data->data[i])[j].get<TypeIndexToType<TypeIndex::UInt8>>(), buf);
+            }
+        }
+    }
```

Comment:
It's better to `serialize/deserialize` the whole column with `{serialize/deserialize}BinaryBulkWithMultipleStreams` methods. You can get serialization object from data type by `getDefaultSerialization` method.

Here are some examples:
https://github.com/ClickHouse/ClickHouse/blob/182ba1c6059da5c1c3314b31b97e8491dd34563f/src/Formats/NativeReader.cpp#L68-L84
https://github.com/ClickHouse/ClickHouse/blob/182ba1c6059da5c1c3314b31b97e8491dd34563f/src/Formats/NativeWriter.cpp#L49-L65